src/Schema/Context.php

```diff
@@ -36,11 +30,23 @@ public function __construct(Request $request)
      */
     public function user(): ?Authenticatable
     {
-        return $this->user;
+        return $this->request->user(AuthServiceProvider::guard());
     }
 
     public function request(): Request
     {
         return $this->request;
     }
+
+    public function __get(string $name): mixed
+    {
+        return method_exists($this, $name)
+            ? $this->{$name}()
+            : $this->{$name};
+    }
+
+    public function __isset(string $name): bool
+    {
+        return method_exists($this, $name) || property_exists($this, $name);
```

Comment:
```suggestion
        return $this->__get($name) !== null;
```

And make `__get()` not throw on undefined keys. `isset()` is supposed to return `false` if the value is `null`.

Reply:
I dug deeper on failing SubscriptionTest::testWithGuard().

Turns out resolving user 'on-demand' can't work properly as expected. (Even if you comment the logout part in the test). Using magical `__get()`, and this attempt to mitigate data_get and @inject directive issue with an `__isset()` is probably not the way to go.

I'll give some more try with a $context->setUser().